src/helpers/HDWalletSafeProvider.js

```diff
@@ -0,0 +1,213 @@
+const assert = require('assert')
+const truffleContract = require('truffle-contract')
+const TruffleHDWalletProvider = require('truffle-hdwallet-provider')
+const Web3 = require('web3')
+
+const Logger = require('./Logger')
+const sendTxWithUniqueNonce = require('./sendTxWithUniqueNonce')
+
+const environment = process.env.NODE_ENV
+const isLocal = environment === 'local'
+
+const logger = new Logger('dx-service:helpers:HDWalletSafeProvider')
+
+// Disable the nonce lock:
+//    - LOCAL: Disabled by default (true)
+//    - DEV/PRE/PRO: Enabled by default (false)
+//    - Modificable by env var DISABLE_NONCE_LOCK
+const NONCE_LOCK_DISABLED = process.env.DISABLE_NONCE_LOCK === 'true' || isLocal
+
+class HDWalletSafeProvider extends TruffleHDWalletProvider {
+
+  constructor ({
+    mnemonic,
+    url,
+    addressIndex = 0,
+    numAddresses = 5,
+    shareNonce = true,
+    blockForNonceCalculation = 'pending',
+    operator, // User account that is gonna send transactions
+    safeAddress,
+    safeModuleAddress,
+    safeModuleMode = 'complete', // Use complete safe-module by default
+    defaultGas = 300000
+  }) {
+    assert(mnemonic, '"mnemonic" is mandatory')
+    assert(url, '"url" is mandatory')
+    assert(safeAddress, '"safeAddress" is mandatory for running in Safe mode')
+    assert(safeModuleAddress, '"safeModuleAddress" is mandatory for running in Safe mode')
+
+    super(mnemonic, url, addressIndex, numAddresses, shareNonce)
+    this._blockForNonceCalculation = blockForNonceCalculation
+    this._defaultGas = defaultGas
+    this._operator = operator || this.addresses[addressIndex] // use 1st account in HDWallet
+    this._safeAddress = safeAddress
+    this._safeModuleAddress = safeModuleAddress
+    this._safeModuleMode = safeModuleMode
+    this._web3 = new Web3(this)
+  }
+
+  loadSafeModule () {
+    let moduleABI
+    if (this.safeModuleMode == 'seller') {
+      moduleABI = require('gnosis-safe-modules/build/contracts/DutchXSellerModule.json')
+    } else {
+      moduleABI = require('gnosis-safe-modules/build/contracts/DutchXCompleteModule.json')
+    }
+
+    this._safeModule = truffleContract(moduleABI).at(this._safeModuleAddress)
```

Comment:
Just curious :) 

Usually I need to set the provider when I instantiate a contract using truffle contract. You didn't need to do that?
Sth like `this._safeModule.setProvider(this)`